controllers/nugus_controller/nugus_controller.cpp

```diff
@@ -15,38 +15,64 @@
  * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
  *
  * Copyright 2021 NUbots <nubots@newcastle.edu.au>
+ *
+ * Portions of this code is taken from the official Webots RoboCup player controller
+ * 2021 by Cyberbotics.
  */
 
-// You may need to add webots include files such as
-// <webots/DistanceSensor.hpp>, <webots/Motor.hpp>, etc.
-// and/or add some other includes
+
 #include <cstdlib>
 #include <iostream>
 #include <memory>
 #include <string>
 #include <vector>
+
+#include <webots/Accelerometer.hpp>
+#include <webots/Camera.hpp>
+#include <webots/Gyro.hpp>
+#include <webots/Motor.hpp>
+#include <webots/Node.hpp>
+#include <webots/PositionSensor.hpp>
 #include <webots/Robot.hpp>
+#include <webots/TouchSensor.hpp>
 
-#include "RobotControl.pb.h"
+#include "messages.pb.h"
 
 #include "utility/tcp.hpp"
 
+#ifdef TURBOJPEG
+#include <turbojpeg.h>
+#else
+#include <jpeglib.h>
+#endif
+
 using namespace utility::tcp;
+using controller::nugus::ActuatorRequests;
+using controller::nugus::CameraExposure;
+using controller::nugus::CameraQuality;
+using controller::nugus::MotorPID;
+using controller::nugus::SensorMeasurements;
+using controller::nugus::SensorTimeStep;
+using controller::nugus::AccelerometerMeasurement;
+using controller::nugus::CameraMeasurement;
+using controller::nugus::GyroMeasurement;
+using controller::nugus::PositionSensorMeasurement;
+using controller::nugus::BumperMeasurement;
+using controller::nugus::ForceMeasurement;
+using controller::nugus::Force3DMeasurement;
+using controller::nugus::Vector3;
 
 class NUgus : public webots::Robot {
 public:
     NUgus(const int& time_step, const int& server_port)
         : time_step(time_step), server_port(server_port), tcp_fd(create_socket_server(server_port)) {
-            send(tcp_fd, "Welcome", 8, 0);
+        send(tcp_fd, "Welcome", 8, 0);
     }
```

Comment:
A blocking constructor just feels wrong.

Reply:
You mean the send function being inside the constructor? Should it be removed?

Reply:
The function `create_socket_server` blocks until a connection is formed.

Reply:
See #17